### PR TITLE
Added `paddle.tensor.logic.logical_or` frontend

### LINKED ISSUE #15951: logical_or

### PR DESCRIPTION
**Pull request by mobley-trent**
Closes #15951

### CODE REVIEW

**Comment by mobley-trent** [comment]
The failing test case indicates that the function requires two tensors, yet I specifically set the testing strategy to produce 2 arrays with the `num_arrays` parameter in the testing strategy. Why is this happening ?

**Comment by mobley-trent** [comment]
Hi @AnnaTz do you mind reviewing this PR ?

File: ivy_tests/test_ivy/test_frontends/test_paddle/test_tensor/test_paddle_logic.py [hunk 1]
```diff
@@ -212,3 +213,29 @@ def test_paddle_equal_all(
         x=x[0],
         y=x[1],
     )
+
+
+# logical_or
+@handle_frontend_test(
+    fn_tree="paddle.logical_or",
+    dtype_and_x=helpers.dtype_and_values(available_dtypes=("bool",), num_arrays=2),
```
Comment (AnnaTz) [inline ivy_tests/test_ivy/test_frontends/test_paddle/test_tensor/test_paddle_logic.py:221]: Our policy is to set `available_dtypes` to "valid" and refine the supported dtypes in the function's decorator.

File: ivy/functional/frontends/paddle/tensor/logic.py [hunk 2]
```diff
@@ -75,3 +75,13 @@ def less_than(x, y, /, *, name=None):
 @to_ivy_arrays_and_back
 def less_equal(x, y, /, *, name=None):
     return ivy.less_equal(x, y)
+
+
+@with_unsupported_dtypes(
+    {"2.4.2 and below": ("bool", "uint8", "int8", "int16", "complex64", "complex128")},
```
Comment (AnnaTz) [inline ivy/functional/frontends/paddle/tensor/logic.py:81]: Are you sure `bool` is unsupported?

File: ivy/functional/frontends/paddle/tensor/logic.py [hunk 3]
```diff
@@ -75,3 +75,13 @@ def less_than(x, y, /, *, name=None):
...
+    "paddle",
+)
+@to_ivy_arrays_and_back
+@handle_out_argument
+def logical_or(x, y, /, *, name=None, out=None):
+    return ivy.logical_or(x, y, out=out)
```
Comment (AnnaTz) [inline ivy/functional/frontends/paddle/tensor/logic.py:87]: There is an assertion error that the results don't match the ground truth. There is probably some difference between ivy's function and paddle's. I suggest looking at the documentation and at ivy's paddle backend implementation to spot the differences.

**Review by AnnaTz — CHANGES_REQUESTED** [review]

File: ivy/functional/frontends/paddle/tensor/logic.py (hunk 3 above)
Comment (mobley-trent) [reply ivy/functional/frontends/paddle/tensor/logic.py:87]: I didn't see the error in the CI. It must have been another function

File: ivy_tests/test_ivy/test_frontends/test_paddle/test_tensor/test_paddle_logic.py [hunk 4]
```diff
@@ -212,3 +213,31 @@ def test_paddle_equal_all(
         x=x[0],
         y=x[1],
     )
+
+
+# logical_or
+@handle_frontend_test(
+    fn_tree="paddle.logical_or",
+    dtype_and_x=helpers.dtype_and_values(
+        available_dtypes=("valid",), num_arrays=2, min_num_dims=1, max_num_dims=3
```
Comment (AnnaTz) [inline ivy_tests/test_ivy/test_frontends/test_paddle/test_tensor/test_paddle_logic.py:222]: `available_dtypes=helpers.get_dtypes("valid")`

**Review by AnnaTz — CHANGES_REQUESTED** [review]

File: ivy/functional/frontends/paddle/tensor/logic.py (hunk 3 above)
Comment (mobley-trent) [reply ivy/functional/frontends/paddle/tensor/logic.py:101]: The ground truth assertions for `paddle_logical_or` are passing in the CI.

**Pull request merged**